uniform-tensors.rkt

```diff
@@ -0,0 +1,47 @@
+#lang racket/base
```

Comment:
C1: So suggestion for reorganizing this PR in to multiple nested branches. This will allow us to keep the merge-tree clean.
I know it's a bit of a pain to do this with such a large PR, but magit is your friend. Get familiar with cherry-picking and breaking up bigger commits into smaller ones. 

So I'm thinking a nesting structure like this: Nesting means the nested branch is a branch of the outer branch (not the main branch, unless that is the outer branch. 
```
malt/main
   |--> add-fixes-to-learner
            |--> add-fixes-to-nested
                      |--> add-fixes-to-flat
                              |--> add-fixes-to-malted
                                      |--> add-uniform-tensors
                                              |--> add-accelerated-tensors
                                                      |--> add-lazy-tensors
```

Reply:
This is only a suggested structure. Feel free to organize as you wish.

Reply:
I have added 8 new PRs. I hope it's organized as you suggested. Unfortunately, moving around commits seems to have introduced two new failing test cases.

Reply:
One thing which came to my mind while creating the "fixes" PRs is that I have added some fixes to the accelerated implementation which should have been reflected in the flat implementation, but haven't. I'll add an issue regarding this once the PRs are merged.